internal/entitlement/connector.go

```diff
@@ -88,7 +88,29 @@ func (c *entitlementConnector) CreateEntitlement(ctx context.Context, input Crea
 		return nil, err
 	}
 
-	ent, err := c.entitlementRepo.CreateEntitlement(ctx, input)
+	var usagePeriod *UsagePeriodWithNextReset
+	if input.UsagePeriod != nil {
+		nextResetAt, err := input.UsagePeriod.GetNextReset()
+		if err != nil {
+			return nil, err
+		}
+		usagePeriod = &UsagePeriodWithNextReset{
+			UsagePeriod: *input.UsagePeriod,
+			NextReset:   nextResetAt,
+		}
+	}
+
+	ent, err := c.entitlementRepo.CreateEntitlement(ctx, CreateEntitlementRepoInputs{
```

Comment:
why are the interfaces different now (repo create vs create)?

Reply:
As the calculations should happen inside the entitlement connector not at the repo side.